drivers/lm75/lm75.c

```diff
@@ -0,0 +1,329 @@
+        return LM75_ERROR_I2C;
+    }
+
+    /* read the device ID register of the TMP1075 sensor to see if the sensor truly is a TMP1075 */
+    if (IS_USED(MODULE_TMP1075)) {
+        uint16_t deid = 0;
+        if (i2c_read_regs(I2C_BUS, I2C_ADDR, TMP1075_DEVICE_ID_REG, &deid, 2, 0) != 0) {
+            LOG_INFO("Error reading device ID\n");
+         }
+
+        deid = (uint16_t)htons(deid);
+        /* checks if the device ID corresponds to the TMP1075 sensor
+         * and extends the parameter configuration if so */
+        if (deid == 0x7500) {
+            puts("Device is a TMP1075");
+            config |= (params->conv_rate << 5) | (params->one_shot_mode << 7);
+            dev->lm75_params.res = &tmp1075_res;
+        }
+        else {
+            puts("Device ID Register doesnt match");
+            return LM75_ERROR;
+        }
+    }
+    else if (IS_USED(MODULE_LM75A)) {
+        puts("Device is an LM75A");
+        dev->lm75_params.res = &lm75a_res;
+    }
+    else {
+        LOG_ERROR("Device not supported\n");
+    }
+
+    /* write the config byte into the configuration register */
+    if (i2c_write_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+    return LM75_SUCCESS;
+}
+
+int lm75_get_temperature(lm75_t *dev, int *temperature) {
+
+    int16_t temp;
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+    /* read the temperature register */
+    if (i2c_read_regs(I2C_BUS, I2C_ADDR, LM75_TEMP_REG, &temp, 2, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+    i2c_release(I2C_BUS);
+
+    /* since the value is read in little endian, it must be converted to big endian
+    * then it must be multiplied by its resolution
+    * and then shifted so only the correct number of bits is used */
+    *temperature = (int16_t)htons(temp) * dev->lm75_params.res->temp_res \
+                    >> dev->lm75_params.res->temp_shift;
+
+    return LM75_SUCCESS;
+}
+
+int lm75_set_temp_limits(lm75_t *dev, int temp_hyst, int temp_os, gpio_cb_t cb, void *arg) {
+
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+    uint8_t config, polarity = 1;
+    if (i2c_read_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, &config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+    /* check os pin polarity */
+    if ((config & 0b00000100) == 0) {
+        polarity = 0;
+    }
+
+    /* Enable OS interrupt */
+    gpio_init_int(LM75_PARAM_INT, GPIO_IN, polarity, cb, arg);
+
+    if (temp_hyst >= temp_os) {
+        LOG_ERROR("THYST must be lower than TOS\n");
+        return LM75_ERROR;
+    }
+    int16_t temp_hyst_short, temp_os_short;
+    /* getting into the correct precision value in units of 10 */
+        temp_hyst *= dev->lm75_params.res->os_mult / 1000;
+        temp_os   *= dev->lm75_params.res->os_mult / 1000;
```

Comment:
```suggestion
    temp_hyst *= dev->lm75_params.res->os_mult / 1000;
    temp_os   *= dev->lm75_params.res->os_mult / 1000;
```

Also shouldn't this be

```suggestion
    temp_hyst = (temp_hyst * dev->lm75_params.res->os_mult) / 1000;
    temp_os   = (temp_os * dev->lm75_params.res->os_mult) / 1000;
```

to avoid rounding errors?